experiments/opencog/pattern_matcher_vqa/tbd_cog/tbd_helpers.py

```diff
@@ -0,0 +1,253 @@
+                                              filter_argument,
+                                              exec_out_sub])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_relate(atomspace, relate_argument, exec_out_sub):
+    schema = atomspace.add_node(types.GroundedSchemaNode, "py:relate")
+    lst = atomspace.add_link(types.ListLink, [atomspace.add_node(types.ConceptNode, relate_argument),
+                                              exec_out_sub])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_intersect(atomspace, arg0, arg1):
+    schema =  atomspace.add_node(types.GroundedSchemaNode, "py:intersect")
+    lst = atomspace.add_link(types.ListLink, [arg0,
+                                              arg1])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_bind_link(atomspace, eval_link, inheritance):
+    """
+    Build bind link from ExecutionOutput or Evaluation links and inheritance links
+
+    :param atomspace: AtomSpace
+    :param eval_link: ExecutionOutput or EvaluationLink
+    :param inheritance: Set[InheritanceLink]
+    :return: BindLink
+    """
+    varlist = []
+    for inh in inheritance:
+        for atom in inh.get_out():
+            if atom.type == types.VariableNode:
+                varlist.append(atom)
+    variable_list = atomspace.add_link(types.VariableList, varlist)
+    list_link = atomspace.add_link(types.ListLink, varlist + [eval_link])
+    conj = atomspace.add_link(types.AndLink, [*inheritance, eval_link])
+    bind_link = atomspace.add_link(types.BindLink, [variable_list, conj, list_link])
+    return bind_link
+
+
+def build_scene(atomspace, scene):
+    schema =  atomspace.add_node(types.GroundedSchemaNode, "py:init_scene")
+    lst = atomspace.add_link(types.ListLink, [scene])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_same(atomspace, same_argument, exec_out_sub):
+    schema = atomspace.add_node(types.GroundedSchemaNode, "py:same")
+    lst = atomspace.add_link(types.ListLink, [atomspace.add_node(types.ConceptNode, same_argument),
+                                              exec_out_sub])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def return_prog(atomspace, commands, inheritance_set=None):
+    """
+    Convert tbd program to atomese
+
+    :param atomspace: Atomspace
+    :param commands: list
+    :param inheritance_set: set
+        Inhertiance links constructed during conversion
+    :return: Tuple[ExecutionOutputLink, list, set]
+        program in form of execution output link, unprocessed part of the program, set of inheritance links
+    """
+    current, rest = commands[0], commands[1:]
+    if inheritance_set is None:
+        inheritance_set = set()
+    scene = atomspace.add_node(types.VariableNode, "$Scene")
+    if current.startswith('query'):
```

Comment:
It is worth to add examples of the commands to make it clear how parsing works for people who are not familiar with CLEVR

Reply:
Done